vault/resource_secrets_sync_azure_destination.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	syncutil "github.com/hashicorp/terraform-provider-vault/internal/sync"
+)
+
+const (
+	fieldKeyVaultURI = "key_vault_uri"
+	fieldCloud       = "cloud"
+	azureSyncType    = "azure-kv"
+)
+
+var azureSyncWriteFields = []string{
+	fieldKeyVaultURI,
+	fieldCloud,
+	consts.FieldClientSecret,
+	consts.FieldClientID,
+	consts.FieldTenantID,
+}
+
+var azureSyncUpdateFields = []string{
+	consts.FieldClientSecret,
+	consts.FieldClientID,
+}
+
+var azureSyncReadFields = []string{
+	fieldCloud,
+	consts.FieldClientID,
+	consts.FieldTenantID,
+}
+
+func azureSecretsSyncDestinationResource() *schema.Resource {
+	return &schema.Resource{
```

Comment:
I think it's missing the `provider.MustAddSecretsSyncCloudSchema(&schema.Resource{` to have the custom_tag & secret_template_name fields.

Reply:
Thanks for the catch! Updated